vsphere/resource_vsphere_snapshot.go

```diff
@@ -0,0 +1,186 @@
+		return fmt.Errorf(" Error While Creating the Task for Create Snapshot: %s", err)
+	}
+	log.Printf("[INFO] Task created for Create Snapshot: %v", task)
+	err = task.Wait(context.TODO())
+
+	if err != nil {
+		log.Printf("[ERROR] Error While waiting for the Task for Create Snapshot: %v", err)
+		return fmt.Errorf(" Error While waiting for the Task for Create Snapshot: %s", err)
+	}
+	log.Printf("[INFO] Create Snapshot completed %v", d.Get("snapshot_name").(string))
+	log.Println("[INFO] Managed Object Reference: " + taskInfo.Result.(types.ManagedObjectReference).Value)
+	d.SetId(taskInfo.Result.(types.ManagedObjectReference).Value)
+	return nil
+}
+
+func resourceVSphereSnapshotDelete(d *schema.ResourceData, meta interface{}) error {
+	vm, err := findVM(d, meta)
+	if err != nil {
+		return fmt.Errorf("Error while getting the VirtualMachine :%s", err)
+	}
+	resourceVSphereSnapshotRead(d, meta)
+	if d.Id() == "" {
+		log.Printf("[ERROR] Error While finding the Snapshot: %v", err)
+		return nil
+	}
+	log.Printf("[INFO] Deleting snapshot with name: %v", d.Get("snapshot_name").(string))
+	var consolidate_ptr *bool
+	var remove_children bool
+
+	if v, ok := d.GetOk("consolidate"); ok {
+		consolidate := v.(bool)
+		consolidate_ptr = &consolidate
+	} else {
+
+		consolidate := true
+		consolidate_ptr = &consolidate
+	}
+	if v, ok := d.GetOk("remove_children"); ok {
+		remove_children = v.(bool)
+	} else {
+
+		remove_children = false
+	}
+
+	task, err := vm.RemoveSnapshot(context.TODO(), d.Id(), remove_children, consolidate_ptr)
+
+	if err != nil {
+		log.Printf("[ERROR] Error While Creating the Task for Delete Snapshot: %v", err)
+		return fmt.Errorf("Error While Creating the Task for Delete Snapshot: %s", err)
+	}
+	log.Printf("[INFO] Task created for Delete Snapshot: %v", task)
+
+	err = task.Wait(context.TODO())
+	if err != nil {
+		log.Printf("[ERROR] Error While waiting for the Task of Delete Snapshot: %v", err)
+		return fmt.Errorf("Error While waiting for the Task of Delete Snapshot: %s", err)
+	}
+	log.Printf("[INFO] Delete Snapshot completed %v", d.Get("snapshot_name").(string))
+
+	return nil
+}
+
+func resourceVSphereSnapshotRead(d *schema.ResourceData, meta interface{}) error {
+	vm, err := findVM(d, meta)
+	if err != nil {
+		return fmt.Errorf("Error while getting the VirtualMachine :%s", err)
+	}
+	snapshot, err := vm.FindSnapshot(context.TODO(), d.Id())
+
+	if err != nil {
+		if strings.Contains(err.Error(), "No snapshots for this VM") || strings.Contains(err.Error(), "snapshot \""+d.Get("snapshot_name").(string)+"\" not found") {
+			log.Printf("[ERROR] Error While finding the Snapshot: %v", err)
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[ERROR] Error While finding the Snapshot: %v", err)
+		return fmt.Errorf("Error while finding the Snapshot :%s", err)
+	}
+	log.Printf("[INFO] Snapshot found: %v", snapshot)
+	return nil
+}
+
+func findVM(d *schema.ResourceData, meta interface{}) (*object.VirtualMachine, error) {
```

Comment:
Once this is changed to find by UUID, can you move this function to a new file called `virtual_machine_helper.go`? I want to try and start ensuring that all helpers are organized by their inparticular MO.

Also, make sure the function name is self-documenting - in this case, it would be `virtualMachineFromUUID`.